### PR TITLE
I2C master mode - basic driver

### LINKED ISSUE #3: I2C - master mode basic driver
No extra features included. The most basic setup for HTS221 Relative humidity and temperature sensor communication.

- max 255 bytes
- only master mode
- no IT
- no DMA

All these features need follow-up in the feature.

Initial code base already on main branch.

### PR DESCRIPTION
**Pull request by MateuszMyalski**
Added I2C in master mode Tx/Rx.
Max 256 bytes.
Pooling mode.
Tested on HTS221 IC
Closes #3

### CODE REVIEW

File: apps/stm32u5/u585xx_bring-up/hal/include/i2c.h [hunk 1]
```diff
@@ -1,18 +1,59 @@
 #ifndef I2C_H
 #define I2C_H
-
+#include <stddef.h>
 #include "hal_common.h"
 
 typedef enum { I2C_mode_slave, I2C_mode_master, I2C_mode_SMBUS_master, I2C_mode_SMBUS_slave } I2C_mode_t;
 
 typedef enum { I2C_speed_10kHz, I2C_speed_100kHz, I2C_speed_400kHz, I2C_speed_1MHz } I2C_speed_t;
```
Comment (bgs94) [inline apps/stm32u5/u585xx_bring-up/hal/include/i2c.h:9]: Shouldn't wee keep each enum element in a separate line?

**Review by bgs94 — COMMENTED** [review]
General question to periphs.c: why do we have separate functions for i2c1 and i2c2? Could we on the other hand use single function with parameter that will tell which peripheral instance to configure?

File: apps/stm32u5/u585xx_bring-up/hal/include/i2c.h (hunk 1 above)
Comment (MateuszMyalski) [reply apps/stm32u5/u585xx_bring-up/hal/include/i2c.h:9]: We can decide. Only the newest version of clang-format support this feature. We can try to upgrade or keep it as clang propose as default.

**Comment by MateuszMyalski** [comment]
> General question to periphs.c: why do we have separate functions for i2c1 and i2c2? Could we on the other hand use single function with parameter that will tell which peripheral instance to configure?

Good idea, this can be joined together!

**Comment by bgs94** [comment]
Looks good to me so far!

**Comment by MateuszMyalski** [comment]
@bgs94 Comments from patch 4 fixed. Ready for another check :) 
image

**Review by bgs94 — APPROVED** [review]
Nice!

**Pull request merged**